src/model.c

```diff
@@ -206,7 +207,7 @@ int RAI_ModelInit(RedisModuleCtx* ctx) {
   return RedisAI_ModelType != NULL;
 }
 
-RAI_Model *RAI_ModelCreate(RAI_Backend backend, RAI_Device device, int64_t deviceid,
+RAI_Model *RAI_ModelCreate(RAI_Backend backend, RAI_Device device, int64_t deviceid, const char* devicestr,
```

Comment:
I personally think we should defer parsing the `devicestr` into `RAI_Device` and `deviceid` until we hit the backend. So, remove them from `RAI_ModelCreate` and `model_create`.
Each backend will be responsible for parsing the backend and stating that it's valid.

Reply:
@lantiga please revise the latest commit. the backends are now the ones responsible for parsing the devicestr.